pkg/eventprocessor/impl/executor/impl/executor_test.go

```diff
@@ -0,0 +1,265 @@
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	require.NoError(t, ex.Close(ctx))
+}
+
+func TestMultiEventTxnBlock(t *testing.T) {
+	t.Parallel()
+
+	// The test setup is executing a block that has the following transactions:
+	// - Txn 1, successful with two events: CREATE a table and INSERT a row.
+	// - Txn 2, failed with two events: CREATE a table and INSERT a row. INSERT has invalid SQL.
+	// - Txn 3, successful with one event: INSERT a row in the table created in Txn 1.
+	//
+	// Checks after block execution:
+	// 1) Txn 1 side-effects are seen in the db.
+	// 2) Txn 2 side-effects aren't seen in the db. In particular, the table created should have been rollbacked.
+	// 3) Txn 3 side-effects are seen in the db. Txn 2 failing should be isolated and have no effect in Txn 3 execution.
+	ctx := context.Background()
+	ex, dbURI := newExecutor(t, 0)
+
+	bs, err := ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+
+	// Txn 1
+	{
+		eventCreateTable := &ethereum.ContractCreateTable{
+			TableId:   big.NewInt(100),
+			Owner:     common.HexToAddress("0xb451cee4A42A652Fe77d373BAe66D42fd6B8D8FF"),
+			Statement: "create table bar_1337 (zar text)",
+		}
+		eventInsertRow := &ethereum.ContractRunSQL{
+			IsOwner:   true,
+			TableId:   eventCreateTable.TableId,
+			Statement: "insert into bar_1337_100 values ('txn 1')",
+			Policy: ethereum.ITablelandControllerPolicy{
+				AllowInsert:      true,
+				AllowUpdate:      true,
+				AllowDelete:      true,
+				WhereClause:      "",
+				WithCheck:        "",
+				UpdatableColumns: nil,
+			},
+		}
+		res, err := bs.ExecuteTxnEvents(ctx, eventfeed.TxnEvents{
+			Events: []interface{}{eventCreateTable, eventInsertRow},
+		})
+		require.NoError(t, err)
+		require.Nil(t, res.Error)
+		require.Nil(t, res.ErrorEventIdx)
+		require.Equal(t, eventCreateTable.TableId.Int64(), res.TableID.ToBigInt().Int64())
+	}
+	// Txn 2
+	{
+		eventCreateTable := &ethereum.ContractCreateTable{
+			TableId:   big.NewInt(101),
+			Owner:     common.HexToAddress("0xb451cee4A42A652Fe77d373BAe66D42fd6B8D8FF"),
+			Statement: "create table foo_1337 (fooz text)",
+		}
+		eventInsertRow := &ethereum.ContractRunSQL{
+			IsOwner:   true,
+			TableId:   eventCreateTable.TableId,
+			Statement: "insert into foo_1337 values ('txn 1', 'wrong # of columns')",
+			Policy: ethereum.ITablelandControllerPolicy{
+				AllowInsert:      true,
+				AllowUpdate:      true,
+				AllowDelete:      true,
+				WhereClause:      "",
+				WithCheck:        "",
+				UpdatableColumns: nil,
+			},
+		}
+		res, err := bs.ExecuteTxnEvents(ctx, eventfeed.TxnEvents{
+			Events: []interface{}{eventCreateTable, eventInsertRow},
+		})
+		require.NoError(t, err)
+		// This Txn should fail.
+		require.NotNil(t, res.Error)
+		require.Equal(t, 1, *res.ErrorEventIdx)
+	}
+	// Txn 3
+	{
+		// We can leverage helper for single event txn.
+		assertExecTxnWithRunSQLEvents(t, bs, []string{"insert into bar_1337_100 values ('txn 3')"})
```

Comment:
I've created some `assertExecTxnWithXXXX(...)` helpers for each event type. These helpers simplify executing transactions with single events to avoid boilerplate in 80% of tests.

In more complex scenarios, such as the two Txn1 and Txn2 scopes above, you can use the full API as used there.